packages/server/graphql/mutations/helpers/upgradeToTeamTier.ts

```diff
@@ -67,6 +73,7 @@ const upgradeToTeamTier = async (
   ])
 
   await Promise.all([setUserTierForOrgId(orgId), setTierForOrgUsers(orgId)])
+  return clientSecret
```

Comment:
With the current approach we upgrade tier to `team` before the successful payment.

I can interrupt payment or 3D secure will fail and I still get the upgrade

https://www.loom.com/share/05260018c4264c57b7d278b412ac8aa1

Reply:
@nickoferrall sorry, missed it

Reply:
All good! I wanted to share this PR even though `UpgradeToTeamTierMutation` needs refactoring because I've already made several changes, and that refactor will add quite a bit more code